src/tsolvers/lasolver/LASolver.cc

```diff
@@ -288,6 +288,13 @@ std::unique_ptr<Tableau::Polynomial> LASolver::expressionToLVarPoly(PTRef term)
 //
 // Returns internalized reference for the term
 LVRef LASolver::registerArithmeticTerm(PTRef expr) {
+    if (logic.isTimesNonlin(expr)) throw NonLinException(logic.pp(expr));
```

Comment:
I don't think this is the right place to check.
It seems to me that a better place would be in `getLAVar_single` which now `asserts` it gets linear term.
There we can check if we are getting nonlin times and throw the exception.

Reply:
Fixed